composer/datasets/streaming/writer.py

```diff
@@ -90,7 +92,9 @@ def __init__(self,
                  dirname: str,
                  fields: List[str],
                  shard_size_limit: int = 1 << 24,
-                 compression: Optional[str] = default_compression) -> None:
+                 compression: Optional[str] = default_compression,
+                 remote: Union[ObjectStore, str, None] = None,
+                 timeout: Optional[float] = None) -> None:
```

Comment:
This is more of a design question, but I'm not thrilled about propagating the timeout setting all the through. My thinking is that:

1. Not all object stores support timeout -- currently, just boto3 does.
2. There are other config settings of equal importance (namly, retries), and propagating all these settings through would complicate public API -- if one wants this amount of customization (i.e. more params than can be specified via the URI), then the can pass an already-constructed `ObjectStore` instance through.

(This comment applies to the other functions that take a `timeout`, except for `download_or_wait`.)

Thoughts?

Reply:
agreed, this makes sense. Let's have the connection strings and remove the `timeout` argument for now -- if the user wants to specify it manually, they can pass an `ObjectStore` as an argument.